src/riak_kv_qry_compiler.erl

```diff
@@ -482,40 +489,27 @@ extract_stateful_functions2({{window_agg_fn, FnName}, _} = Function, FinaliserLe
 
 %% Only change the start order if the query has descending fields, otherwise
 %% the natural order is correct.
-maybe_fix_start_order(false, W, _, _) ->
+maybe_fix_start_order(false, W) ->
     W;
-maybe_fix_start_order(true, W, Mod, LK) ->
-    fix_start_order(W, Mod, LK).
+maybe_fix_start_order(true, W) ->
+    fix_start_order(W).
 
 %%
-fix_start_order(W, Mod, LK) ->
+fix_start_order(W) ->
     {startkey, StartKey0} = lists:keyfind(startkey, 1, W),
     {endkey, EndKey0} = lists:keyfind(endkey, 1, W),
-    case is_start_key_greater(Mod, LK, StartKey0, EndKey0) of
-        true ->
-            W;
-        false ->
-            %% Swap the start/end keys so that the backends will
-            %% scan over them correctly.  Likely cause is a descending
-            %% timestamp field.
-            W1 = lists:keystore(startkey, 1, W, {startkey, EndKey0}),
-            W2 = lists:keystore(endkey, 1, W1, {endkey, StartKey0}),
-            W1 = lists:keystore(startkey, 1, W, {startkey, EndKey0}),
-            W2 = lists:keystore(endkey, 1, W1, {endkey, StartKey0}),
-            %% start inclusive defaults true, end inclusive defaults false
-            W3 = lists:keystore(start_inclusive, 1, W2,
-                                {start_inclusive, proplists:get_value(end_inclusive, W, false)}),
-            _W4 = lists:keystore(end_inclusive, 1, W3,
-                                 {end_inclusive, proplists:get_value(start_inclusive, W2, true)})
-    end.
-
-%%
-is_start_key_greater(Mod, LK, StartKey0, EndKey0) ->
-    StartVals = [{N, V} || {N, _T, V} <- StartKey0],
-    EndVals = [{N, V} || {N, _T, V} <- EndKey0],
-    StartKey = riak_ql_ddl:make_key(Mod, LK,  StartVals),
-    EndKey = riak_ql_ddl:make_key(Mod, LK,  EndVals),
-    (StartKey < EndKey).
+    %% Swap the start/end keys so that the backends will
+    %% scan over them correctly.  Likely cause is a descending
+    %% timestamp field.
+    W1 = lists:keystore(startkey, 1, W, {startkey, EndKey0}),
+    W2 = lists:keystore(endkey, 1, W1, {endkey, StartKey0}),
+    W1 = lists:keystore(startkey, 1, W, {startkey, EndKey0}),
+    W2 = lists:keystore(endkey, 1, W1, {endkey, StartKey0}),
```

Comment:
My excuse is that my keyboard's delete button is broken ;) (it actually is). Fixed in a64c5a50a4a0393d958eeecc34210d95fe5324ae.